libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -1090,8 +1103,10 @@ AP_GPS_UBLOX::_parse_gps(void)
     case MSG_TIMEGPS:
         Debug("MSG_TIMEGPS");
         _check_new_itow(_buffer.timegps.itow);
-        if (_buffer.timegps.valid & UBX_TIMEGPS_VALID_WEEK_MASK)
+        if (_buffer.timegps.valid & UBX_TIMEGPS_VALID_WEEK_MASK) {
             state.time_week = _buffer.timegps.week;
+        }
+        haveTimeGPSMsg = true;
```

Comment:
Might as well remove this variable, we already make the assumption in the SOL rate, which would be vulnerable as well.